extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/AbstractTokensProducer.java

```diff
@@ -1,27 +1,37 @@
 package io.quarkus.oidc.client.runtime;
 
+import java.util.Objects;
+import java.util.Optional;
+
 import javax.annotation.PostConstruct;
 import javax.inject.Inject;
 
 import org.eclipse.microprofile.config.inject.ConfigProperty;
 
 import io.quarkus.oidc.client.OidcClient;
+import io.quarkus.oidc.client.OidcClients;
 import io.quarkus.oidc.client.Tokens;
 import io.smallrye.mutiny.Uni;
 
 public abstract class AbstractTokensProducer {
 
     @Inject
-    OidcClient oidcClient;
+    OidcClients oidcClients;
```

Comment:
Is there a risk of double `close` here, once when `oidcClients.close()` is called and once when `oidcClient.close()` is called ?

Reply:
I don't think so: the `OidcClient` is `private` and never explicitly closed (and won't be closed by CDI as it's not injected here)

Reply:
@fwippe, ok, makes sense; by the way related to these `OidcClients` and `OidcClient` - I believe it would be cheaper to use `Arc.container` to get `OidcClients` in the post-construct method - as it is only needed to initialize `OidcClient` - so keeping the injection point for `OidcClients` is not really needed

Reply:
OK, switched to `Arc.container` invocatoin.